src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/StatelessMarshallingStrategy.cs

```diff
@@ -71,18 +72,26 @@ public IEnumerable<StatementSyntax> GenerateMarshalStatements(StubIdentifierCont
                 yield break;
             }
 
-            // <nativeIdentifier> = <convertToUnmanaged>;
-            var assignment = AssignmentExpression(
-                    SyntaxKind.SimpleAssignmentExpression,
-                    IdentifierName(nativeIdentifier),
-                    convertToUnmanaged);
-
+            ExpressionSyntax assignment;
 
-            if (unmanagedType is PointerTypeInfo pointer)
+            // For some of our exception marshallers, our marshaller returns nint for pointer types.
+            // As a result, we need to insert a cast here in case we're in that scenario (which we can't detect specifically).
+            if (unmanagedType is PointerTypeInfo ptrType)
+            {
+                // <nativeIdentifier> = (<nativeType>)<convertToUnmanaged>;
+                assignment = AssignmentExpression(SyntaxKind.SimpleAssignmentExpression,
+                    IdentifierName(nativeIdentifier),
+                    CastExpression(ptrType.Syntax, convertToUnmanaged));
+            }
+            else
             {
-                var rewriter = new PointerNativeTypeAssignmentRewriter(assignment.Right.ToString(), (PointerTypeSyntax)pointer.Syntax);
-                assignment = (AssignmentExpressionSyntax)rewriter.Visit(assignment);
+                // <nativeIdentifier> = <convertToUnmanaged>;
+                assignment = AssignmentExpression(
+                        SyntaxKind.SimpleAssignmentExpression,
+                        IdentifierName(nativeIdentifier),
+                        convertToUnmanaged);
             }
+
```

Comment:
Do we need anything like this for native marshalling?

Reply:
We only need this for unmarshalling because we only hit it in a specific (due to us artificially resolving the marshaller) scenario. We don't put ourselves into that situation in the managed->native case.